exercises/practice/mazy-mice/src/test/java/MazeGeneratorTest.java

```diff
@@ -0,0 +1,193 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MazeGeneratorTest {
+    private static final char EMPTY_CELL = ' ';
+    private static final Set<Character> ALLOWED_SYMBOLS = Set.of(
+            EMPTY_CELL, // space
+            '┌', // box drawings light down and right
+            '─', // box drawings light horizontal
+            '┬', // box drawings light down and horizontal
+            '┐', // box drawings light down and left
+            '│', // box drawings light vertical
+            '└', // box drawings light up and right
+            '┴', // box drawings light up and horizontal
+            '┘', // box drawings light up and left
+            '├', // box drawings light vertical and right
+            '┤', // box drawings light vertical and left
+            '┼', // box drawings light vertical and horizontal
+            '⇨'  // rightwards white arrow
+    );
+    private static final char VISITED_CELL = '.';
+    private static final Dimensions RECTANGLE = new Dimensions(6, 18);
+    private MazeGenerator sut;
+
+    @Before
+    void SetUp() {
+        sut = new MazeGenerator();
+    }
+
+    @Test
+    void mazeIsNotNull() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+
+        assertThat(maze)
+                .as("The maze is not null.")
+                .isNotNull();
+
+        Arrays.stream(maze)
+                .map(String::new)
+                .forEach(System.out::println);
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    void the_dimensions_are_correct() {
+        var maze = sut.generatePerfectMaze(RECTANGLE);
+
+        assertThat(maze)
+                .as("The dimensions of the maze are correct.")
+                .hasDimensions(RECTANGLE.height(), RECTANGLE.width());
+    }
```

Comment:
Please bear in mind that students are shown the source code of each failing test, and that it helps them if that snippet contains as much information as possible. Right now, a student would have to look up what the value of `RECTANGLE` is. IMO it's clearer if the test specifies the values inline instead of reusing a static field:

``` java
class MazeGeneratorTest {
    @Test
    void the_dimensions_are_correct() {
        var rows = 6;
        var columns = 18;
        var maze = sut.generatePerfectMaze(new Dimensions(rows, columns));

        var expectedHeight = 13;
        var expectedWidth = 37;
        assertThat(maze)
                .as("The dimensions of the maze are correct.")
                .hasDimensions(expectedHeight, expectedWidth);
    }
}
```

Another benefit of this approach is that it takes a little bit less effort to keep the test cases in line with the problem-specifications, as I noticed that you have opened a PR for this exercise there too.